setup-pre-commit/action.yml

```diff
@@ -0,0 +1,50 @@
+---
+name: Run pre-commit
+description: |
+  This actions sets up Python, the pre-commit tool and additional tools required
+  for various hooks. It then runs pre-commit against the changed files. This
+  actions expects checkouts with depth 0.
+inputs:
+  python-version:
+    description: Python version to install
+    default: "3.12"
+  rust:
+    description: Whether to install the Rust toolchain (and which version to use)
+  hadolint:
+    description: Whether to install hadolint (and which version to use)
+runs:
+  using: composite
+  steps:
+    - name: Setup Python
+      uses: actions/setup-python@f677139bbe7f9c59b41e40162b753c062f5d49a3 # v5.2.0
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: Setup Rust Toolchain
+      uses: dtolnay/rust-toolchain@7b1c307e0dcbda6122208f10795a713336a9b35a
+      if: ${{ inputs.rust }}
+      with:
+        toolchain: ${{ inputs.rust }}
+        components: rustfmt,clippy
```

Comment:
Nothing to change now, but maybe this should be overridable. For example, somewhere, we needed `rust-src` too. Maybe we want to add more components one day?

Reply:
Added in 69b8e50